### PR TITLE
fix: use module_to_chunk for dynamic imports with tree-shaking

### LINKED ISSUE #6513: [Bug]: AWS SDK required code is tree-shaken away / throws error
### Reproduction link or steps

Rolldown is either erroring or tree-shaking away parts of the [AWS JavaScript SDK v3](https://github.com/aws/aws-sdk-js-v3). I think it's something to do with dynamic imports but I haven't had a chance to dive much deeper!

Here's a minimal repo which reproduces the issue: https://github.com/cogwirrel/bugs/tree/rolldown-dynamic-import-aws-sdk

### What is expected?

All used code to be included in the bundle.

### What is actually happening?

The `fromIni` method from `@aws-sdk/credential-provider-ini` gets stripped out when `inlineDynamicImports` is `true`, causing an error at runtime when creating AWS service clients.

Rolldown resolves to the `dist-cjs/index.js` file in node modules for this dependency, which looks like:

```js
'use strict';

var sharedIniFileLoader = require('@smithy/shared-ini-file-loader');
var propertyProvider = require('@smithy/property-provider');
var client = require('@aws-sdk/core/client');

const resolveCredentialSource = (credentialSource, profileName, logger) => {
    const sourceProvidersMap = {
        EcsContainer: async (options) => {
            const { fromHttp } = await import('@aws-sdk/credential-provider-http');
            const { fromContainerMetadata } = await import('@smithy/credential-provider-imds');
            logger?.debug("@aws-sdk/credential-provider-ini - credential_source is EcsContainer");
            return async () => propertyProvider.chain(fromHttp(options ?? {}), fromContainerMetadata(options))().then(setNamedProvider);
        },
...
const fromIni = (_init = {}) => async ({ callerClientConfig } = {}) => {
    ...
};

exports.fromIni = fromIni;
```

But rolldown's output strips out everything below the initial requires:

```
//#region node_modules/.pnpm/@aws-sdk+credential-provider-ini@3.908.0/node_modules/@aws-sdk/credential-provider-ini/dist-cjs/index.js
var require_dist_cjs$3 = /* @__PURE__ */ __commonJS({ "node_modules/.pnpm/@aws-sdk+credential-provider-ini@3.908.0/node_modules/@aws-sdk/credential-provider-ini/dist-cjs/index.js": ((exports) => {
	var sharedIniFileLoader$1 = require_dist_cjs$21();
	var propertyProvider$1 = require_dist_cjs$30();
	var client = (init_client(), __toCommonJS$8(client_exports));
}) });

//#endregion
```

This causes an error at runtime when creating SDK clients as the `fromIni` method is missing.

When `inlineDynamicImports` is omitted, Rolldown throws an error:

```
thread '<unnamed>' panicked at crates/rolldown/src/module_finalizers/mod.rs:972:83:
no entry found for key
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

thread '<unnamed>' panicked at crates/rolldown/src/module_finalizers/mod.rs:972:83:
no entry found for key

 ERROR  Panic in async function
```

When I disable tree shaking it works as expected.

### System Info

```Shell
System:
    OS: macOS 15.7.1
    CPU: (16) arm64 Apple M4 Max
    Memory: 735.69 MB / 128.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 22.14.0 - /Users/jack/.nvm/versions/node/v22.14.0/bin/node
    Yarn: 1.22.22 - /Users/jack/.nvm/versions/node/v22.14.0/bin/yarn
    npm: 10.9.2 - /Users/jack/.nvm/versions/node/v22.14.0/bin/npm
    pnpm: 10.15.1 - /Users/jack/.nvm/versions/node/v22.14.0/bin/pnpm
    bun: 1.1.45 - /Users/jack/.bun/bin/bun
  Browsers:
    Chrome: 141.0.7390.77
    Firefox: 143.0.1
    Safari: 26.0.1
  npmPackages:
    rolldown: 1.0.0-beta.43 => 1.0.0-beta.43
```

### Any additional comments?

I'm using rolldown as the bundler for the [Nx Plugin for AWS](https://awslabs.github.io/nx-plugin-for-aws/) - this vends "generators" which set up AWS projects for you.

When working with AWS you use the SDK a lot, so it's important that this gets bundled correctly. I can work around this with a plugin that disables tree-shaking for the SDK, but it'd be much better not to have to vend this workaround in our generators.

### PR DESCRIPTION
**Pull request by bishonen**
Fixes #6513 (which at this point is still broken)

This PR fixes a panic that occurs when tree-shaking is enabled with dynamic imports.

The issue was that the code was using `entry_module_to_entry_chunk` to look up chunks for dynamically imported modules. This map only contains entry point modules, causing a panic when dynamically imported modules (like AWS SDK modules) weren't entry points.

The fix changes to use `module_to_chunk` which contains all modules mapped to their containing chunks, with proper None handling for edge cases.

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
Please add a test case

File: crates/rolldown/src/module_finalizers/mod.rs [hunk 1]
```diff
@@ -1044,7 +1044,16 @@ impl<'me, 'ast> ScopeHoistingFinalizer<'me, 'ast> {
...
-          let importee_chunk_id = self.ctx.chunk_graph.entry_module_to_entry_chunk[&importee_id];
+          // Try entry_module_to_entry_chunk first (for code-split dynamic imports),
+          // fall back to module_to_chunk (for modules bundled in same chunk)
+          let importee_chunk_id = self
+            .ctx
+            .chunk_graph
+            .entry_module_to_entry_chunk
+            .get(&importee_id)
+            .copied()
+            .or_else(|| self.ctx.chunk_graph.module_to_chunk[importee_id])
```
Comment (IWANABETHATGUY) [inline crates/rolldown/src/module_finalizers/mod.rs:1055]: Use `self.ctx.chunk_graph.module_to_chunk[importee_id]` directly

**Comment by IWANABETHATGUY** [comment]
image
Please confirm that you have resolved the issue. I tried your PR with #6513 reproduction; it still panic.

**Comment by bishonen** [comment]
I believe(d) it did back when I committed it (alongside the aws-sdk errors). Perhaps I was wrong, as trying it again today still shows the error.

The aws-sdk errors have been solved one way or another in version 46, so this fix would be unnecessary.

I'll close this request because of this. The inline error will still persist and would require a new pr.
